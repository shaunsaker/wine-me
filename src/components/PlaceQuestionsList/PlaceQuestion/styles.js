import { StyleSheet } from 'react-native';

import styleConstants from '../../../styleConstants';

const styles = StyleSheet.create({
  container: {
    paddingVertical: styleConstants.dimensions.spacing.vertical,
    paddingHorizontal: styleConstants.dimensions.spacing.horizontal,
  },
  questionText: {
    ...styleConstants.fonts.types.title,
    textAlign: 'center',
    marginBottom: styleConstants.dimensions.spacing.vertical * 2,
  },
  starRatingContainer: {
    alignItems: 'center',
  },
  textInputContainer: {},
  buttonsContainer: {
    marginBottom: styleConstants.dimensions.spacing.vertical * -1, // remove last button margin
  },
  buttonContainer: {
    marginBottom: styleConstants.dimensions.spacing.vertical,
  },
});

export default styles;
