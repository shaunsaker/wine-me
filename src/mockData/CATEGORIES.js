const CATEGORIES = {
  wine: {
    order: 1,
    validation: {
      type: 'number',
      min: 1,
      max: 5,
    },
    name: 'Wine',
    question_text: 'How would you rate the wine?',
    is_rating: true,
  },
  child_friendly: {
    order: 2,
    validation: {
      type: 'string',
    },
    name: 'Child friendly',
    question_text: 'Was it child friendly?',
    options: ['Yes', 'No'],
  },
  price: {
    order: 3,
    validation: {
      type: 'number',
      min: 0,
    },
    name: 'Price',
    question_text: 'What did the wine tasting cost?',
  },
};

export default CATEGORIES;
