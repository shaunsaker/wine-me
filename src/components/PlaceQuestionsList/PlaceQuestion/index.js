import React from 'react';
import PropTypes from 'prop-types';
import { View, Text } from 'react-native';

import styles from './styles';

import StarRating from '../../StarRating';
import TextInput from '../../TextInput';
import Button from '../../Button';

const propTypes = {
  category: PropTypes.shape({
    name: PropTypes.string,
    options: PropTypes.arrayOf(PropTypes.string),
    validation: PropTypes.shape({
      type: PropTypes.string,
    }),
  }).isRequired,
  value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  handleSetValue: PropTypes.func,
};

const defaultProps = {};

const PlaceQuestion = ({ category, value, handleSetValue }) => {
  /*
    If category.is_rating
      Display StarRating
    Else if options
      Display two buttons
    Else
      Display text input
  */

  let inputComponent;

  if (category.is_rating) {
    inputComponent = (
      <View style={styles.starRatingContainer}>
        <StarRating rating={value} handlePress={handleSetValue} />
      </View>
    );
  } else if (category.options) {
    inputComponent = (
      <View style={styles.buttonsContainer}>
        {category.options.map((option) => {
          return (
            <View key={option} style={styles.buttonContainer}>
              <Button
                primary={option === value}
                text={option}
                handlePress={() => handleSetValue(option)}
              />
            </View>
          );
        })}
      </View>
    );
  } else {
    inputComponent = (
      <View style={styles.textInputContainer}>
        <TextInput
          placeholder={category.name}
          keyboardType={category.validation && category.validation.type === 'number' && 'numeric'}
          handleChange={(string) => handleSetValue(Number(string))}
        />
      </View>
    );
  }

  return (
    <View style={styles.container}>
      <Text style={styles.questionText}>{category.question_text}</Text>

      {inputComponent}
    </View>
  );
};

PlaceQuestion.propTypes = propTypes;
PlaceQuestion.defaultProps = defaultProps;

export default PlaceQuestion;
