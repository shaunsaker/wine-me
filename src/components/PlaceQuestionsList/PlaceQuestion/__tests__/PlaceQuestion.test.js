import React from 'react';
import renderer from 'react-test-renderer';

import PlaceQuestion from '..';
import CATEGORIES from '../../../../mockData/CATEGORIES';

describe('PlaceQuestion', () => {
  describe('renders', () => {
    it('renders with minimum required props', () => {
      const category = CATEGORIES.wine;
      const component = renderer.create(<PlaceQuestion category={category} />);

      expect(component).toMatchSnapshot();
    });

    describe('renders the rating state', () => {
      const category = CATEGORIES.wine;

      it('with a value', () => {
        const value = 4;
        const component = renderer.create(<PlaceQuestion category={category} value={value} />);

        expect(component).toMatchSnapshot();
      });

      it('without a value', () => {
        const value = null;
        const component = renderer.create(<PlaceQuestion category={category} value={value} />);

        expect(component).toMatchSnapshot();
      });
    });

    describe('renders the options state', () => {
      const category = CATEGORIES.child_friendly;

      it('with a value', () => {
        const value = 'Yes';
        const component = renderer.create(<PlaceQuestion category={category} value={value} />);

        expect(component).toMatchSnapshot();
      });

      it('without a value', () => {
        const value = null;
        const component = renderer.create(<PlaceQuestion category={category} value={value} />);

        expect(component).toMatchSnapshot();
      });
    });

    describe('renders the text input state', () => {
      const category = CATEGORIES.price;

      it('with a value', () => {
        const value = 100;
        const component = renderer.create(<PlaceQuestion category={category} value={value} />);

        expect(component).toMatchSnapshot();
      });

      it('without a value', () => {
        const value = null;
        const component = renderer.create(<PlaceQuestion category={category} value={value} />);

        expect(component).toMatchSnapshot();
      });
    });
  });
});
