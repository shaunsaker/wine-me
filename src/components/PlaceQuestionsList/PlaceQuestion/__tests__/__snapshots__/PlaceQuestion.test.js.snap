// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceQuestion renders renders the options state with a value 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 15,
      "paddingVertical": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#212121",
        "fontFamily": "Nunito-SemiBold",
        "fontSize": 24,
        "marginBottom": 20,
        "textAlign": "center",
      }
    }
  >
    Was it child friendly?
  </Text>
  <View
    style={
      Object {
        "marginBottom": -10,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#E23E57",
            "borderColor": null,
            "borderRadius": 5,
            "borderWidth": 0,
            "elevation": 2,
            "opacity": 1,
            "paddingHorizontal": 15,
            "paddingVertical": 10,
            "shadowColor": "#212121",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.17,
            "shadowRadius": 2,
          }
        }
        testID="largeButton"
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#212121",
                "fontFamily": "Nunito-SemiBold",
                "fontSize": 17,
              },
              Object {
                "color": "#FFFFFF",
              },
              undefined,
            ]
          }
        >
          Yes
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": null,
            "borderRadius": 5,
            "borderWidth": 0,
            "elevation": 2,
            "opacity": 1,
            "paddingHorizontal": 15,
            "paddingVertical": 10,
            "shadowColor": "#212121",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.17,
            "shadowRadius": 2,
          }
        }
        testID="largeButton"
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#212121",
                "fontFamily": "Nunito-SemiBold",
                "fontSize": 17,
              },
              false,
              undefined,
            ]
          }
        >
          No
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`PlaceQuestion renders renders the options state without a value 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 15,
      "paddingVertical": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#212121",
        "fontFamily": "Nunito-SemiBold",
        "fontSize": 24,
        "marginBottom": 20,
        "textAlign": "center",
      }
    }
  >
    Was it child friendly?
  </Text>
  <View
    style={
      Object {
        "marginBottom": -10,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": null,
            "borderRadius": 5,
            "borderWidth": 0,
            "elevation": 2,
            "opacity": 1,
            "paddingHorizontal": 15,
            "paddingVertical": 10,
            "shadowColor": "#212121",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.17,
            "shadowRadius": 2,
          }
        }
        testID="largeButton"
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#212121",
                "fontFamily": "Nunito-SemiBold",
                "fontSize": 17,
              },
              false,
              undefined,
            ]
          }
        >
          Yes
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": null,
            "borderRadius": 5,
            "borderWidth": 0,
            "elevation": 2,
            "opacity": 1,
            "paddingHorizontal": 15,
            "paddingVertical": 10,
            "shadowColor": "#212121",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.17,
            "shadowRadius": 2,
          }
        }
        testID="largeButton"
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#212121",
                "fontFamily": "Nunito-SemiBold",
                "fontSize": 17,
              },
              false,
              undefined,
            ]
          }
        >
          No
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`PlaceQuestion renders renders the rating state with a value 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 15,
      "paddingVertical": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#212121",
        "fontFamily": "Nunito-SemiBold",
        "fontSize": 24,
        "marginBottom": 20,
        "textAlign": "center",
      }
    }
  >
    How would you rate the wine?
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="starRating.button.0"
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Object {
                "color": "#E23E57",
                "fontSize": 48,
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="starRating.button.1"
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Object {
                "color": "#E23E57",
                "fontSize": 48,
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="starRating.button.2"
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Object {
                "color": "#E23E57",
                "fontSize": 48,
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="starRating.button.3"
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Object {
                "color": "#E23E57",
                "fontSize": 48,
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="starRating.button.4"
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Object {
                "color": "#E23E57",
                "fontSize": 48,
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`PlaceQuestion renders renders the rating state without a value 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 15,
      "paddingVertical": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#212121",
        "fontFamily": "Nunito-SemiBold",
        "fontSize": 24,
        "marginBottom": 20,
        "textAlign": "center",
      }
    }
  >
    How would you rate the wine?
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="starRating.button.0"
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Object {
                "color": "#E23E57",
                "fontSize": 48,
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="starRating.button.1"
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Object {
                "color": "#E23E57",
                "fontSize": 48,
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="starRating.button.2"
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Object {
                "color": "#E23E57",
                "fontSize": 48,
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="starRating.button.3"
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Object {
                "color": "#E23E57",
                "fontSize": 48,
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="starRating.button.4"
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Object {
                "color": "#E23E57",
                "fontSize": 48,
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`PlaceQuestion renders renders the text input state with a value 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 15,
      "paddingVertical": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#212121",
        "fontFamily": "Nunito-SemiBold",
        "fontSize": 24,
        "marginBottom": 20,
        "textAlign": "center",
      }
    }
  >
    What did the wine tasting cost?
  </Text>
  <View
    style={Object {}}
  >
    <TextInput
      allowFontScaling={true}
      keyboardType="numeric"
      onChangeText={[Function]}
      placeholder="Price"
      placeholderTextColor="#757575"
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "borderColor": "#E0E0E0",
            "borderRadius": 5,
            "borderWidth": 1,
            "color": "#212121",
            "fontFamily": "Nunito-Regular",
            "fontSize": 17,
            "paddingHorizontal": 15,
            "paddingVertical": 10,
            "textAlign": "center",
            "textAlignVertical": "center",
          },
          undefined,
          undefined,
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;

exports[`PlaceQuestion renders renders the text input state without a value 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 15,
      "paddingVertical": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#212121",
        "fontFamily": "Nunito-SemiBold",
        "fontSize": 24,
        "marginBottom": 20,
        "textAlign": "center",
      }
    }
  >
    What did the wine tasting cost?
  </Text>
  <View
    style={Object {}}
  >
    <TextInput
      allowFontScaling={true}
      keyboardType="numeric"
      onChangeText={[Function]}
      placeholder="Price"
      placeholderTextColor="#757575"
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "borderColor": "#E0E0E0",
            "borderRadius": 5,
            "borderWidth": 1,
            "color": "#212121",
            "fontFamily": "Nunito-Regular",
            "fontSize": 17,
            "paddingHorizontal": 15,
            "paddingVertical": 10,
            "textAlign": "center",
            "textAlignVertical": "center",
          },
          undefined,
          undefined,
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;

exports[`PlaceQuestion renders renders with minimum required props 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 15,
      "paddingVertical": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#212121",
        "fontFamily": "Nunito-SemiBold",
        "fontSize": 24,
        "marginBottom": 20,
        "textAlign": "center",
      }
    }
  >
    How would you rate the wine?
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="starRating.button.0"
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Object {
                "color": "#E23E57",
                "fontSize": 48,
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="starRating.button.1"
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Object {
                "color": "#E23E57",
                "fontSize": 48,
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="starRating.button.2"
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Object {
                "color": "#E23E57",
                "fontSize": 48,
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="starRating.button.3"
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Object {
                "color": "#E23E57",
                "fontSize": 48,
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="starRating.button.4"
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Object {
                "color": "#E23E57",
                "fontSize": 48,
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;
